pkg/operator/ceph/csi/spec.go

```diff
@@ -662,3 +688,265 @@ func (r *ReconcileCSI) validateCSIVersion(ownerInfo *k8sutil.OwnerInfo) (*CephCS
 	}
 	return version, nil
 }
+
+type migrationJobReport struct {
+	nodeName string
+	err      error
+}
+
+func (r *ReconcileCSI) setupCSINetwork() error {
+	// Setup of the CSI Network depends on if multus is being used.
+	var isMultusApplied bool
+	var publicNetwork string
+	var ok bool
+
+	cephClusters := &cephv1.CephClusterList{}
+	err := r.client.List(r.opManagerContext, cephClusters, &client.ListOptions{})
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debug("no ceph cluster found not deploying ceph csi driver")
+			return nil
+		}
+		// Error reading the object - requeue the request.
+		return errors.Wrap(err, "failed to list ceph clusters")
+	}
+
+	for _, cephCluster := range cephClusters.Items {
```

Comment:
Detecting if multus is needed looks like in two different places. How about factoring to a common method?